core/get_ratings.py

```diff
@@ -1,148 +1,141 @@
+    looser_new_rating = update_rating(game_ratings.opponent, score.looser,
+                                      lost_expected, ball, set_counts[players.looser])
+    return GameRatings(winner_new_rating, looser_new_rating)
 
 
 def update_ratings_with_game(ratings, game, set_counts):
-    winner = game.Winner
-    looser = game.Looser
-    sets = list(map(int, game.Score.split(':')))
-
-    winner_sets = np.max(sets)
-    looser_sets = np.min(sets)
-    ball_coeff = constants.BALL_COEFFS[game.Ball]
-    winner_rating = ratings.loc[winner].Rating
-    looser_rating = ratings.loc[looser].Rating
-
-    winner_new_rating, looser_new_rating = apply_elo_law(winner, looser, winner_sets, looser_sets,
-                                                         ball_coeff, winner_rating, looser_rating, set_counts)
+    players = Players(game.Winner, game.Looser)
+    score = ScoreInSets(*sorted(map(int, game.Score.split(':')), reverse=True))
+    game_ratings = GameRatings(ratings.loc[players.winner].Rating, ratings.loc[players.looser].Rating)
+
+    game_ratings_after = apply_elo_law(players, score, game_ratings, game.Ball, set_counts)
     new_ratings = ratings.copy(deep=True)
-    new_ratings.loc[winner].Rating = winner_new_rating
-    new_ratings.loc[looser].Rating = looser_new_rating
-    total_sets = winner_sets + looser_sets
-    set_counts[winner] += total_sets
-    set_counts[looser] += total_sets
+    new_ratings.loc[players.winner].Rating = game_ratings_after.my
+    new_ratings.loc[players.looser].Rating = game_ratings_after.opponent
+    set_counts[players.winner] += score.winner + score.looser
+    set_counts[players.looser] += score.winner + score.looser
     return new_ratings
 
 
 def count_ratings(date):
-    game_data_filepath = os.path.join(repo_root_dir, 'data/games.csv')
+    game_data_filepath = os.path.join(REPO_ROOT_DIR, 'data/games.csv')
     games = pd.read_csv(game_data_filepath)
     sorted_games = games.sort_values(['Date', 'Game of the day'])
     proper_games = sorted_games[sorted_games['Date'] <= date]
+
     all_players = list(set(proper_games['Winner'].tolist() + proper_games['Looser'].tolist()))
     data = {'Rating': [constants.DEFAULT_RATING for i in range(len(all_players))]}
     start_ratings = pd.DataFrame(data, index=all_players)
     n_games = proper_games.shape[0]
     set_counts = {name: 0 for name in all_players}
     ratings_log = [(proper_games.loc[0]['Date'], start_ratings)]
+
     for i in range(n_games):
         new_ratings = update_ratings_with_game(ratings_log[-1][1], proper_games.loc[i], set_counts)
         new_ratings = new_ratings.sort_values(['Rating'], ascending=False)
         ratings_log.append((proper_games.loc[i]['Date'], new_ratings.copy(deep=True)))
     return ratings_log[-1][1], ratings_log, set_counts
 
 
+def get_previous_rating(name, ratings_log):
+    for i in range(len(ratings_log)):
+        cur_day, cur_ratings = ratings_log[i]
+        if cur_day == ratings_log[-1][0]:
```

Comment:
what's the point?
`cur_day == last_day`?

Reply:
Yes, but I need first occurrence of last_day to find previous match day

Reply:
Is it possibly to refactor this code in more understandable way?

Reply:
I think, yes